common/src/bytecode.rs

```diff
@@ -207,4 +210,34 @@ mod tests {
     fn empty_hex_bytecode_is_empty() {
         assert!(Bytecode::from_hex_str("0x").unwrap().is_empty());
     }
+
+    #[test]
+    fn to_fixed_hex() {
+        for (value, expected) in &[
+            (
+                "0x0000000000000000000000000000000000000000",
+                "0000000000000000000000000000000000000000",
+            ),
+            (
+                "0x0102030405060708091020304050607080900001",
+                "0102030405060708091020304050607080900001",
+            ),
+            (
+                "0x9fac3b52be975567103c4695a2835bba40076da1",
+                "9fac3b52be975567103c4695a2835bba40076da1",
+            ),
+        ] {
+            let value: Address = value[2..].parse().unwrap();
+            assert_eq!(super::to_fixed_hex(&value), *expected);
+        }
+    }
+
+    #[test]
+    fn string_replacement_replaces_multiple_occurrences() {
+        // This is important for Bytecode::link.
```

Comment:
I would prefer if this method test `Bytecode::Link` directly. And maybe a second test to verify the branch where the library is not needed?